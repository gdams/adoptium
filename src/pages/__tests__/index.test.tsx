import React from 'react';
import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe';
import Index from '../index';

describe('Index page', () => {
  it('renders correctly', () => {
    const { container } = render(<Index />);
    // eslint-disable-next-line
    const pageContent = container.querySelector('main');

    expect(pageContent).toMatchSnapshot();
  });

  it('has no accessibility violations', async () => {
    const { container } = render(<Index />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
