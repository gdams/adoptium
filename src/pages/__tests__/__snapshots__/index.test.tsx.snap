// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Index page > renders correctly 1`] = `
<main
  style="padding-top: 100px;"
>
  <div
    class="main-banner"
    id="top"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-7"
        >
          <div
            class="caption header-text"
          >
            <p
              class="h4 tag-line"
            >
              Text
              <br />
              Text
            </p>
            <p>
              Text
            </p>
            <div
              class="hide-on-mobile"
              style="max-width: 550px;"
            >
              <h2
                class="fw-light mt-3 text-pink medium"
              >
                Download Temurin™
              </h2>
              <div
                class="btn-group-vertical mx-auto col-12"
              >
                <a
                  class="btn btn-lg btn-primary mt-3 py-3 text-white"
                  href="/temurin/releases"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
                    />
                  </svg>
                   Latest LTS releases
                </a>
                <a
                  class="btn btn-outline-dark mt-3"
                  href="/temurin/archive"
                >
                  Text
                   
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M32 448c0 17.7 14.3 32 32 32h384c17.7 0 32-14.3 32-32V160H32v288zm160-212c0-6.6 5.4-12 12-12h104c6.6 0 12 5.4 12 12v8c0 6.6-5.4 12-12 12H204c-6.6 0-12-5.4-12-12v-8zM480 32H32C14.3 32 0 46.3 0 64v48c0 8.8 7.2 16 16 16h480c8.8 0 16-7.2 16-16V64c0-17.7-14.3-32-32-32z"
                    />
                  </svg>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="services section mb-5"
    id="services"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-6 offset-lg-6"
        >
          <div
            class="row"
          >
            <div
              class="col-lg-12"
            >
              <div
                class="section-heading"
              >
                <h2
                  class="text-pink"
                >
                  Text
                </h2>
                <div
                  class="line-dec"
                />
                <p>
                  Text
                </p>
              </div>
            </div>
            <div
              class="col-lg-6 col-sm-6"
            >
              <div
                class="service-item"
              >
                <svg
                  color="#FF1364"
                  fill="currentColor"
                  height="50"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: rgb(255, 19, 100);"
                  viewBox="0 0 1024 1024"
                  width="50"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M904 512h-56c-4.4 0-8 3.6-8 8v320H184V184h320c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V520c0-4.4-3.6-8-8-8z"
                  />
                  <path
                    d="M355.9 534.9L354 653.8c-.1 8.9 7.1 16.2 16 16.2h.4l118-2.9c2-.1 4-.9 5.4-2.3l415.9-415c3.1-3.1 3.1-8.2 0-11.3L785.4 114.3c-1.6-1.6-3.6-2.3-5.7-2.3s-4.1.8-5.7 2.3l-415.8 415a8.3 8.3 0 0 0-2.3 5.6zm63.5 23.6L779.7 199l45.2 45.1-360.5 359.7-45.7 1.1.7-46.4z"
                  />
                </svg>
                <a
                  href="/join"
                >
                  <p
                    class="h4"
                  >
                    Text
                  </p>
                </a>
                <p>
                  Take the first step in joining this Eclipse Foundation working group by reaching out to our membership team.
                </p>
              </div>
            </div>
            <div
              class="col-lg-6 col-sm-6"
            >
              <div
                class="service-item"
              >
                <svg
                  aria-hidden="true"
                  color="#FF1364"
                  fill="currentColor"
                  height="50"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: rgb(255, 19, 100);"
                  viewBox="0 0 20 20"
                  width="50"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M13 6a3 3 0 11-6 0 3 3 0 016 0zM18 8a2 2 0 11-4 0 2 2 0 014 0zM14 15a4 4 0 00-8 0v3h8v-3zM6 8a2 2 0 11-4 0 2 2 0 014 0zM16 18v-3a5.972 5.972 0 00-.75-2.906A3.005 3.005 0 0119 15v3h-3zM4.75 12.094A5.973 5.973 0 004 15v3H1v-3a3 3 0 013.75-2.906z"
                  />
                </svg>
                <a
                  href="/members"
                >
                  <p
                    class="h4"
                  >
                    Text
                  </p>
                </a>
                <p>
                  The Adoptium Working Group promotes and supports high-quality runtimes and associated technology for use across the Java ecosystem.
                </p>
              </div>
            </div>
            <div
              class="col-lg-6 col-sm-6"
            >
              <div
                class="service-item"
              >
                <svg
                  color="#FF1364"
                  fill="currentColor"
                  height="50"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: rgb(255, 19, 100);"
                  viewBox="0 0 616 512"
                  width="50"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M602 118.6L537.1 15C531.3 5.7 521 0 510 0H106C95 0 84.7 5.7 78.9 15L14 118.6c-33.5 53.5-3.8 127.9 58.8 136.4 4.5.6 9.1.9 13.7.9 29.6 0 55.8-13 73.8-33.1 18 20.1 44.3 33.1 73.8 33.1 29.6 0 55.8-13 73.8-33.1 18 20.1 44.3 33.1 73.8 33.1 29.6 0 55.8-13 73.8-33.1 18.1 20.1 44.3 33.1 73.8 33.1 4.7 0 9.2-.3 13.7-.9 62.8-8.4 92.6-82.8 59-136.4zM529.5 288c-10 0-19.9-1.5-29.5-3.8V384H116v-99.8c-9.6 2.2-19.5 3.8-29.5 3.8-6 0-12.1-.4-18-1.2-5.6-.8-11.1-2.1-16.4-3.6V480c0 17.7 14.3 32 32 32h448c17.7 0 32-14.3 32-32V283.2c-5.4 1.6-10.8 2.9-16.4 3.6-6.1.8-12.1 1.2-18.2 1.2z"
                  />
                </svg>
                <a
                  href="/marketplace"
                >
                  <p
                    class="h4"
                  >
                    Text
                  </p>
                </a>
                <p>
                  The Adoptium Marketplace offers easy access to high-quality OpenJDK Java binaries.
                </p>
              </div>
            </div>
            <div
              class="col-lg-6 col-sm-6"
            >
              <div
                class="service-item"
              >
                <img
                  alt="AQAvit logo"
                  src="/images/aqavit-icon.png"
                  width="50"
                />
                <a
                  href="/aqavit"
                >
                  <p
                    class="h4"
                  >
                    Eclipse AQAvit™
                  </p>
                </a>
                <p>
                  Learn about how the AQAvit project thorougly tests every binary produced by the Temurin project.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
